integration.md

```diff
@@ -0,0 +1,104 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package parent
+
+import (
+	"testing"
+	"time"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
+	"github.com/onsi/gomega"
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	childapis "k8s.io/child/pkg/apis"
+	childv1alpha1 "k8s.io/childrepo/pkg/apis/child/v1alpha1"
+	parentapis "k8s.io/parent/pkg/apis"
+	parentv1alpha1 "k8s.io/parentrepo/pkg/apis/parent/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const timeout = time.Second * 5
+
+var c client.Client
+var expectedRequest = reconcile.Request{NamespacedName: types.NamespacedName{Name: "parent", Namespace: "default"}}
+var childKey = types.NamespacedName{Name: "child", Namespace: "default"}
+
+func TestReconcile(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	// Parent instance to be created.
+	parent := &parentv1alpha1.Parent{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "parent",
+			Namespace: "default",
+		},
+		Spec: metav1.ParentSpec{
+			SomeSpecField:    "SomeSpecValue",
+			AnotherSpecField: "AnotherSpecValue",
+		},
+	}
+
+	// Setup the Manager and Controller. Wrap the Controller Reconcile function
+	// so it writes each request to a channel when it is finished.
+	mgr, err := manager.New(cfg, manager.Options{})
+
+	// Setup Scheme for all resources.
+	if err = parentapis.AddToScheme(mgr.GetScheme()); err != nil {
+		t.Logf("failed to add Parent scheme: %v", err)
+	}
+	if err = childapis.AddToScheme(mgr.GetScheme()); err != nil {
+		t.Logf("failed to add Child scheme: %v", err)
+	}
+
+	// Set up and start test manager.
+	reconciler, err := newReconciler(mgr)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	recFn, requests := SetupTestReconcile(reconciler)
+	g.Expect(add(mgr, recFn)).NotTo(gomega.HaveOccurred())
+	defer close(StartTestManager(mgr, g))
+
+	// Create the Parent object and expect the Reconcile and Child to be created.
+	c = mgr.GetClient()
+	err = c.Create(context.TODO(), parent)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	defer c.Delete(context.TODO(), parent)
+	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
+
+	// Verify Child is created.
+	child := &childv1alpha1.Child{}
+	g.Eventually(func() error { return c.Get(context.TODO(), childKey, child) }, timeout).
+		Should(gomega.Succeed())
+
+	// Manually delete Child since GC isn't enabled in the test control plane.
+	g.Expect(c.Delete(context.TODO(), child)).To(gomega.Succeed())
+}
+```
```

Comment:
Consider removing boilerplate stuff such as import statements. Highlight the stuff which user is tied to reconcile logic and how to test that.

Reply:
I included them to ensure what does `parentapis`, `parentv1alpha1`, `childapis`, and `childv1alpha1` means. I will remove all other import statements except for these 4.